src/vs/workbench/contrib/positronPreview/browser/positronPreviewServiceImpl.ts

```diff
@@ -496,4 +501,35 @@ export class PositronPreviewService extends Disposable implements IPositronPrevi
 		// It's a localhost http or https URL; we can handle it in the viewer.
 		return true;
 	}
+
+	public async openEditor(uri: URI, title?: string): Promise<void> {
```

Comment:
this is where most of the content is; where we create the preview webview and open it in a custom editor. afaict, we are able to use `PreviewUrl` for HTML files and local URLs, since they all will have a uri we can use to open.